lms/product/factory.py

```diff
@@ -11,44 +12,9 @@
 
 def get_product_from_request(request) -> Product:
     """Get the correct product object from the provided request."""
-    ai = request.lti_user.application_instance if request.lti_user else None
+    ai = request.lti_user.application_instance
+    family = Family(request.lti_user.lti.product_family)
```

Comment:
What would happen here if `request.lti_user` is not set? We don't need to check that anymore?

Reply:
I had the same question while refactoring this.

My understanding is that that check was superfluous. If we don't have a lti_user is either:

- It's an LTI launch and we are building the LTIUser at that moment. The code in this PR should build the LTIUser first, populate `product_family` and then arrive here.

- It's an unauthenticated request. Unless I'm missing something important we shouldn't care/query for product in those case. If we found a use case were that's necessary we should return UNKNOWN here but ideally we'd know which those cases are.

Reply:
Gotcha